woocommerce/woocommerce
# Reset a test site

### Changes proposed in this Pull Request:

The goal of this PR is to incorporate resetting the test site into the core E2E tests. It makes the following changes

- introduces a utility flow `withRestApi`
- moves `deleteAllCoupons` and `deleteAllProducts` from the local jest setup to `withRestApi` to allow other projects to take advantage of those functions
- adds an `expect().toBeInRange()` matcher to allow setting up shipping after a reset. See https://github.com/woocommerce/woocommerce-admin/issues/6761
- fixes an issue where the city is duplicated on retest `San FranciscoSan Francisco`
- changes `runSetupWizard()` to not rely on redirect when testing in a reset site

### How to test the changes in this Pull Request:

1. Run `npm install`
2. Run `npx wc-e2e docker:up`
3. Run `npx wc-e2e test:e2e`
4. Tests should pass
5. Run `E2E_RETEST=1 npx wc-e2e test:e2e`
6. Tests should pass 
7. Review changes and additions to package readme and change log files

### Changelog entry

N/A
